Sources/CombineGRPC/Client/BidirectionalStreamingPublisher.swift

```diff
@@ -0,0 +1,38 @@
+// Copyright 2019, ComgineGRPC
+// Licensed under the Apache License, Version 2.0
+
+import Foundation
+import Combine
+import GRPC
+import SwiftProtobuf
+import NIOHPACK
+import NIO
+
+class BidirectionalStreamingPublisher<Request, Response, RequestPublisher>: Publisher
+where RequestPublisher: Publisher, RequestPublisher.Output == Request, RequestPublisher.Failure == Error, Request: Message, Response: Message {
+
+  typealias Output = Response
+  typealias Failure = RPCError
+
+  let rpc: BidirectionalStreamingRPC<Request, Response>
+  let callOptions: CallOptions
+  let requests: RequestPublisher
+
+  init(rpc: @escaping BidirectionalStreamingRPC<Request, Response>, callOptions: CallOptions, requests: RequestPublisher) {
+    self.rpc = rpc
+    self.callOptions = callOptions
+    self.requests = requests
+  }
+
+  func receive<S>(subscriber: S) where S : Subscriber, S.Input == Output, S.Failure == RPCError {
+
+    let buffer = DemandBuffer(subscriber: subscriber)
+
+    let call = rpc(callOptions) { _ = buffer.buffer(value: $0) }
```

Comment:
Note that the server stream may result in OOM if the client can't consume it fast enough.

Reply:
Currently you run into the same problem but at a different point. As SwiftNIO receives these messages it is buffering them. If the I/O thread doesn't keep up you still have an unbounded growing queue.

I would suggest that an enhancement issue be added to allow control of the queue size and how requests are dropped when the queue is full; much like other buffers.